configure.ac

```diff
@@ -4305,6 +4305,10 @@ if test "x$OPENSSL_ENABLED" = "x1" -o "x$GNUTLS_ENABLED" = "x1" \
   SUPPORT_FEATURES="$SUPPORT_FEATURES HTTPS-proxy"
 fi
 
+if test "x$ESNI_ENABLED" = "x1"; then
+  SUPPORT_FEATURES="$SUPPORT_FEATURES ESNI"
+fi
+
```

Comment:
You could squash all your configure commits into a single one to reduce the number of commits and make the PR easier to review...

Reply:
Yes, indeed. Sorry.

Reply:
Do you need me to consolidate this, or is it tolerable as it is just this once?

Reply:
Done. I began to feel that the jumble was hurting even **my** head. I felt it was useful to use two commits (rather than just one) in order to keep handling of the command-line token `--enable-esni` separate from applying the corresponding configuration.